browser/android/brave_sync_worker.cc

```diff
@@ -239,122 +239,58 @@ void BraveSyncWorker::OnResetDone() {
-
-  bool sync_prefs_passphrase_required =
-      service->GetUserSettings()->IsPassphraseRequired();
-
-  if (!first_setup_in_progress) {
-    if (!configuration->encrypt_all) {
-      configuration->encrypt_all = true;
-      configuration->set_new_passphrase = true;
-      DCHECK_NE(passphrase.size(), 0u);
-      configuration->passphrase = passphrase;
-    } else if (sync_prefs_passphrase_required) {
-      configuration->set_new_passphrase = false;
-      DCHECK_NE(passphrase.size(), 0u);
-      configuration->passphrase = passphrase;
-    } else {
-      return false;
-    }
+void BraveSyncWorker::SetEncryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+  DCHECK(!this->passphrase_.empty());
+
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
+  DCHECK(!sync_user_settings->IsPassphraseRequired());
+
+  if (sync_user_settings->IsEncryptEverythingAllowed() &&
+      !sync_user_settings->IsUsingSecondaryPassphrase() &&
+      !sync_user_settings->IsTrustedVaultKeyRequired()) {
+    sync_user_settings->SetEncryptionPassphrase(this->passphrase_);
+    ProfileMetrics::LogProfileSyncInfo(
+        ProfileMetrics::SYNC_CREATED_NEW_PASSPHRASE);
   }
-  return true;
 }
 
-}  // namespace
+void BraveSyncWorker::SetDecryptionPassphrase() {
+  syncer::SyncService* service = GetSyncService();
+  DCHECK(service);
+  DCHECK(service->IsEngineInitialized());
+  DCHECK(!this->passphrase_.empty());
+  syncer::SyncUserSettings* sync_user_settings =
+      GetSyncService()->GetUserSettings();
+  DCHECK(sync_user_settings->IsPassphraseRequired());
+
+  if (sync_user_settings->SetDecryptionPassphrase(this->passphrase_)) {
+    ProfileMetrics::LogProfileSyncInfo(
+        ProfileMetrics::SYNC_ENTERED_EXISTING_PASSPHRASE);
+  }
+}
 
 void BraveSyncWorker::OnStateChanged(syncer::SyncService* sync) {
-  // Fill SyncConfigInfo as it is done in
-  // brave_sync_subpage.js:handleSyncPrefsChanged_ and then configure encryption
-  // as in  PeopleHandler::HandleSetEncryption
-
-  SyncConfigInfo configuration;
-
   syncer::SyncService* service = GetSyncService();
-
   // If the sync engine has shutdown for some reason, just give up
   if (!service || !service->IsEngineInitialized()) {
     VLOG(3) << "[BraveSync] " << __func__ << " sync engine is not initialized";
     return;
   }
 
-  if (!FillSyncConfigInfo(service, &configuration, this->passphrase_)) {
-    VLOG(3) << "[BraveSync] " << __func__
-            << " operations with passphrase are not required";
+  if (this->passphrase_.empty()) {
+    VLOG(3) << "[BraveSync] " << __func__ << " empty passphrase";
     return;
   }
 
-  if (service->GetUserSettings()->IsEncryptEverythingAllowed()) {
-    ProfileMetrics::LogProfileSyncInfo(ProfileMetrics::SYNC_ENCRYPT);
+  if (GetSyncService()->GetUserSettings()->IsPassphraseRequired()) {
```

Comment:
Should we not use service instead of GetSyncService()

Reply:
we should, thanks, fixed at 2bdf046a084cb5445c71d45ef28cfd7cbcab6221